### PR TITLE
[WIP] Add Egress Router Controller

### PR DESCRIPTION
**Pull request by danielmellado**
## Add EgressRouter type and CRD

Signed-off-by: Surya Seetharaman <suryaseetharaman.9@gmail.com>
(cherry picked from commit 0fcbf349c7fc7b01052fd3514e5be0b2dd6084f5)

### CODE REVIEW

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 68b481f74bf55dff29058d99292bdb704fff56a6

**Comment by squeed** [comment]
You need to do `hack/update-goden.sh` to.. do some things. You also need to add another file to the `cp` command.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 541ea9416f31a020cc0e2c2b3dc06d5615164b9b

**Comment by tssurya** [comment]
> You need to do `hack/update-goden.sh` to.. do some things. You also need to add another file to the `cp` command in that script.

ok so for the CRD/deepcopy stuff its this patch: https://github.com/openshift/cluster-network-operator/pull/852.
Dani just has some skeleton of that in this patch which I think he will remove once he re-bases on the CRD changes based on our recent discussion.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 12602c46a9eed4ca1a58bc89bfdb30ad8e7ac77b

File: cmd/cluster-network-operator/main.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 	k8sutil "github.com/openshift/cluster-network-operator/pkg/util/k8s"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"k8s.io/client-go/tools/clientcmd"
+	_ "k8s.io/code-generator"
```
Comment (squeed) [inline cmd/cluster-network-operator/main.go:18]: This should probably go in `dependencymagnet`

Comment (danielmellado) [reply cmd/cluster-network-operator/main.go:18]: it already is there, will just drop this one

**Pull request closed without merging**